app/routes/provider.js

```diff
@@ -19,39 +18,49 @@ export default Ember.Route.extend({
     beforeModel(transition) {
         const {slug} = transition.params.provider;
         const slugLower = (slug || '').toLowerCase();
+        this.store
+            .findAll('preprint-provider')
```

Comment:
Instead of this, just try to load the preprint provider detail page. The api equivalent is, rather than looking through `/v2/preprint_providers/` for `socarxiv`, try to load `/v2/preprint_providers/socarxiv/` and see if it give a 200 or a 400 series error.

The only reason to do this is if you're pre-fetching all of the preprint_provider models for the life of the app, and it's just a few calls once rather than, say, each time you switch providers. But I will let the next reviewer (who should be better versed in ember) decide if this way makes sense or not.